### PR TITLE
Update XL C++14 flag

### PR DESCRIPTION
**Pull request by davidbeckingsale**
Switch from experimental 1y to 14

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
CC @rchen20

File: cmake/SetupCompilerOptions.cmake [hunk 1]
```diff
@@ -268,7 +268,7 @@ if (BLT_CXX_STD)
         blt_append_custom_compiler_flag(
             FLAGS_VAR CMAKE_CXX_FLAGS
             DEFAULT " "
-            XL "-std=c++1y"
+            XL "-std=c++14"
```
Comment (kennyweiss) [inline cmake/SetupCompilerOptions.cmake:271]: @davidbeckingsale -- presumably this is here b/c xl did not previously support the `c++14` flag. 
Do you know which version of XL added support for this?

Bonus points: Do we know if there is a version of CMake that generates the right flag for XL?

**Review by kennyweiss — APPROVED** [review]

File: cmake/SetupCompilerOptions.cmake (hunk 1 above)
Comment (kennyweiss) [reply cmake/SetupCompilerOptions.cmake:271]: Looks like (~2 years ago) CMake 3.10 and XL 16 was still using `c++1y`
https://gitlab.kitware.com/cmake/cmake/-/issues/18071

Comment (kennyweiss) [reply cmake/SetupCompilerOptions.cmake:271]: If I type `man xlC` on our latest version of `xl`, I also still seeing `c++1y` but not `c++14`.

Here's the relevant section:
```
> ml XL/2021.05.19   (latest)
>man xlC
 ...
-std[=<suboption>]
-qlanglvl[=<suboption>]
	   Determines whether source code and compiler options should be checked for conformance to a specific language standard, or subset or superset of a standard.
	   ...

	   (C++) For C++ language programs, the suboptions of -std include:
	   c++98 | c++03
	   gnu++98 | gnu++03
	   c++11 | c++0x
	   gnu++11 | gnu++0x
	   c++1y
```

Comment (rchen20) [reply cmake/SetupCompilerOptions.cmake:271]: I think their documentation might be slightly out-of-date. Although the page below is from September 2020, so the `-std=c++14` flag must be a fairly recent change.

https://www.ibm.com/support/pages/c14-language-standard-fully-supported-xl-cc-linux-1611-fix-pack-8

**Comment by rchen20** [comment]
Hello @davidbeckingsale @kennyweiss, just a friendly reminder to perhaps fix and merge this. On RAJA, we're hoping to have this change in before our imminent release. I would help with the Gitlab failure, but don't have access. Thanks!

**Comment by white238** [comment]
My concern is this breaks previously working versions of XL.  Though I admit I cannot think of a single person that would be on an old XL using the previous compiler flag.

**Comment by kennyweiss** [comment]
Can we do a conditional check on the XL version and add the appropriate flag? (Assuming we still have versions of XL that only support `c++1y`.  I agree with @white238 that we wouldn't want to break backwards compatibility

**Comment by kennyweiss** [comment]
@rchen20 wrote:
>I would help with the Gitlab failure, but don't have access.

It looks like all tests are passing. Where do you see a gitlab failure?

**Comment by white238** [comment]
> 
> 
> @rchen20 wrote:
> 
> > I would help with the Gitlab failure, but don't have access.
> 
> It looks like all tests are passing. Where do you see a gitlab failure?

Gitlab was failing due to some weird system error.  I force reran the pipeline just now.

**Comment by white238** [comment]
Given that XL versions are really hard to figure out:

```
$ module load xl/2019.12.23
$ xlc --version
IBM XL C/C++ for Linux, V16.1.1 (5725-C73, 5765-J13)
Version: 16.01.0001.0006

$ module load xl/2021.05.19
$ xlc --version
IBM XL C/C++ for Linux, V16.1.1 (5725-C73, 5765-J13)
Version: 16.01.0001.0010
```

and the documentation is wrong.  I think I am okay with just merging this as is.  Thoughts @kennyweiss ?

**Comment by kennyweiss** [comment]
That works for me.  We can always revert if there's an insurmountable problem

**Review by white238 — APPROVED** [review]

**Comment by rhornung67** [comment]
> That works for me. We can always revert if there's an insurmountable problem

The XL compiler is an insurmountable problem....

**Pull request merged**